### PR TITLE
Added docs for Products, Stores and Categories APIs

### LINKED ISSUE #12: Provide documentation for Categories, Products and Stores
- Provide an example in the README about how to use the gem.
- Create a new page explaining the usage of the gem for each API endpoint in a friendly way.

### PR DESCRIPTION
**Pull request by SandroDamilano**
Added documentation for the APIs:

- Products
- Stores
- Categories

Also, modified the README and added the `general_overview.md` file for common info across the different APIs.

Closes #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,7 +52,7 @@ To install this gem onto your local machine, run `bundle exec rake install`. To
 
 ## Contributing
 
-Bug reports and pull requests are welcome on GitHub at https://github.com/[USERNAME]/gem-base. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [code of conduct](https://github.com/[USERNAME]/gem-base/blob/master/CODE_OF_CONDUCT.md).
+Bug reports and pull requests are welcome on GitHub at https://github.com/rootstrap/best_buy_ruby. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [code of conduct](https://github.com/[USERNAME]/gem-base/blob/master/CODE_OF_CONDUCT.md).
```
Comment (juanmanuelramallo) [inline README.md:55]:
```suggestion
Bug reports and pull requests are welcome on GitHub at https://github.com/rootstrap/best_buy_ruby. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [code of conduct](https://github.com/rootstrap/best_buy_ruby/blob/master/CODE_OF_CONDUCT.md).
```

File: docs/general_overview.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+- [Products](products_api.md)
+- [Stores](stores_api.md)
+- [Categories](categories_api.md)
+
+In spite of having each their own specification, they all share some basics listed here.
+
+## Common methods
+
+These methods can be used with any of the APIs:
```
Comment (juanmanuelramallo) [inline docs/general_overview.md:13]: How many methods are described below? Isn't only one currently?

File: docs/general_overview.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- `:page`: number of the page of results you’d like returned
+- `:page_size`: amount of results per page (default: 10, max: 100)
+
+Example:
+```ruby
+pagination_params = { page: 3, page_size: 50 }
+
+BestBuy::Stores.new(your_api_key).get_all(pagination: pagination_params)
+```
+
```
Comment (juanmanuelramallo) [inline docs/general_overview.md:68]: Empty line not needed maybe?

File: docs/products_api.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# Products API
+
+## The Product object
+
+Attributes:
+
+- `active`: Identifies if product is currently supported in the BESTBUY.COM catalog
+- `category_path`: A hierarchal view of a product returned as a collection
```
Comment (juanmanuelramallo) [inline docs/products_api.md:8]:
```suggestion
- `category_path`: A hierarchical view of a product returned as a collection
```

File: docs/stores_api.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- `store_id`: Store number
+- `store_type`: Indicates the type of store, and is only present if locationType is ‘Store’. See the [official documentation](https://bestbuyapis.github.io/api-documentation/#common-attributes56) for more details
+- `store_address`: See [The Address object](stores_api.md#the-address-object)
+
+#### `opening_time`
+
+Array of hashes containing info about the opening hours for each day for the next two weeks. It has the following keys:
+
+- `:day`: Name of the day of the week (e.g.: 'Sunday')
+- `:date`: Date of the day (YYYY-MM-DD)
```
Comment (juanmanuelramallo) [inline docs/stores_api.md:22]: Is this a string rather than a Date object?

File: docs/stores_api.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+- `:open`: Opening time (24-hour clock) (`'Close'` if the store is closed)
+- `:close`: Closing time (24-hour clock) (`'Close'` if the store is closed)
```
Comment (juanmanuelramallo) [inline docs/stores_api.md:24]: What do these return? A string with the time? Like `"17:00"` for instance?

File: docs/general_overview.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+In spite of having each their own specification, they all share some basics listed here.
+
+## Common methods
+
+These methods can be used with any of the APIs:
+
+#### `get_all(search_query:, pagination:)`
+_alias: `index`_
+
+Retrieves a collection of items (which ones depends on the API). These items will be filtered by `search_query` and limited by `pagination`.
```
Comment (juanmanuelramallo) [inline docs/general_overview.md:18]:
```suggestion
Retrieves a collection of items (the ones depending on the API). These items will be filtered by `search_query` and limited by `pagination`.
```

File: docs/general_overview.md (hunk 2 above)
Comment (SandroDamilano) [reply docs/general_overview.md:13]: Would `This method` suit better?

File: docs/stores_api.md (hunk 5 above)
Comment (SandroDamilano) [reply docs/stores_api.md:22]: It is! I'll fix the docs now, but I could convert it to a Date object in the future

File: docs/stores_api.md (hunk 6 above)
Comment (SandroDamilano) [reply docs/stores_api.md:24]: Yes! I'll add it to the docs

File: docs/general_overview.md (hunk 7 above)
Comment (SandroDamilano) [reply docs/general_overview.md:18]: What about `(these depend on the API)`?

File: README.md [hunk 8]
```diff
@@ -26,11 +20,29 @@ And then execute:
...
+On a Ruby on Rails project, you can also use a generator to create the gem's initializer:
+
+    $ rails generate best_buy:config
+    
+This will create this file: `config/initializers/best_buy.rb`
 
 ## Usage
 
-TODO: Write usage instructions here
+The gem can be used to access the [APIs](general-overview.md):
```
Comment (fedeagripa) [inline README.md:33]: `to access best buy [APIs]`

**Pull request merged**